core/cli/src/plugin/options.ts

```diff
@@ -59,3 +64,72 @@ export const validatePluginOptions = (logger: Logger, config: ValidPluginsConfig
 
   return invalidOptions
 }
+
+export const substituteOptionTags = (plugin: Plugin, config: ValidPluginsConfig): void => {
+  // foo.bar gets the 'bar' option set for the 'foo' plugin
+  const resolveOptionPath = (optionPath: string): unknown => {
+    const [pluginName, optionName] = optionPath.split('.', 2)
+    return (config.pluginOptions[pluginName] as OptionsForPlugin)?.options[optionName]
+  }
+
+  // throw an error if there are tags in plugin option fields to avoid circular
+  // references
+  const validateTagPath = (path: (string | number)[]) => {
+    if (path[0] === 'options' && path[1] === 'plugins') {
+      const error = new ToolKitError('cannot reference plugin options when specifying options')
+      error.details = `YAML tag referencing options used at path '${path.join('.')}'`
+      throw error
```

Comment:
**suggestion**: use `Validated` for this to avoid whackamole if they've done it multiple times?